src/dbnode/persist/fs/retriever_options.go

```diff
@@ -126,3 +129,13 @@ func (o *blockRetrieverOptions) SetBlockLeaseManager(leaseMgr block.LeaseManager
 func (o *blockRetrieverOptions) BlockLeaseManager() block.LeaseManager {
 	return o.blockLeaseManager
 }
+
+func (o *blockRetrieverOptions) SetQueryStats(value stats.QueryStats) BlockRetrieverOptions {
+	opts := *o
```

Comment:
for my own education, why do you need to dereference the pointer `o` instead of just `o.queryStats = value` ?

Reply:
it's returning a copy

Reply:
This is so we take a copy rather than mutate the underlying actual `o` reference. E.g. https://play.golang.org/p/M8qPBNHO2X1. 

Idiomatic go builder patterns tend to avoid mutating / having a final "build" step at the end. AFAIK, in general it is common in go for people to prefer copying over mutating (e.g. `ctx, cancel := context.WithCancel(ctx)`).